tests/testthat/test-vroom_write.R

```diff
@@ -287,3 +283,32 @@ test_that("vroom_write() always outputs in UTF-8", {
 
   expect_equal(actual_data, expected_data)
 })
+
+test_that("vroom_write() does not quote strings that start with NA", {
+
+  names_df <- tibble::tibble(x = c(NA, "NATHAN", "JOHN", "SALLY", "JONAH"))
+
+  simple_output <- vroom_format(names_df, delim = ",")
+  expect_equal(simple_output,
+               "x\nNA\nNATHAN\nJOHN\nSALLY\nJONAH\n")
+
+  output_unique_NA <- vroom_format(names_df, delim = ",", na = "JON")
+  expect_equal(output_unique_NA,
+               "x\nJON\nNATHAN\nJOHN\nSALLY\nJONAH\n")
+
+
+  output_roundtrip <- vroom(I(vroom_format(names_df)),
+    delim = ",",
+    show_col_types = FALSE
+  )
+  expect_equal(output_roundtrip, names_df)
+})
+
+test_that("vroom-write() no longer quotes NAs", {
+
+  quoted_NA <- tibble::tibble(x = c(NA, "NATHAN", "NA", "SALLY", "JONAH"))
+  output <- vroom_format(quoted_NA)
+
+  expect_equal(output,
+               "x\nNA\nNATHAN\nNA\nSALLY\nJONAH\n")
+})
```

Comment:
Does this test add anything beyond what's covered by the one just above and the one modified near the top of the file?

Once you update the first test, does its purpose and this test's purpose become one and the same?

Reply:
This test felt like it was needed since it's testing that `NA` is no longer quoted when writing output and that `NA` and `"NA"` are equivalent. Where as the tests above are testing the original issue, that strings that start with `NA` are not quoted.

Reply:
This test has a true missing values` NA` and the literal string `"NA"` so in my mind this is testing that the `""` are no longer preserved when writing and the `NA` and `"NA"` are treated the same. The tests directly above are testing that strings that start with "NA" aren't erroneously quoted, which was the original issue. The test further up is testing that the `na` argument can modify values when writing.

There is some overlap and they could all be condensed into a few tests rather than testing the individual pieces.